app/controllers/reports_controller.rb

```diff
@@ -17,4 +18,16 @@ def show
       raise ActiveRecord::RecordNotFound, 'Report not found'
     end
   end
+
+  private
+
+  def require_dashboard_access!
+    return if current_user.can_access_reporting_dashboard?
+
+    raise ForbiddenError, 'Must be a reporting user'
+  end
+
+  def require_report_access!
+    raise ForbiddenError, 'Report access denied' unless current_user.reports.include?(params[:id])
+  end
```

Comment:
In this scenario, I don't think the user doesn't have access, it's that they don't have a report to view. Tim's rework of this made it so that Caseworkers could see their own Caseworker report

Reply:
I checked `main` and this implementation mimics existing behaviour, the difference is the 'access restricted' message is shown rathe than a 'page not found'. Is that incorrect?